gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/internal/MacJarSignFileCopyingProcessor.kt

```diff
@@ -0,0 +1,147 @@
+package org.jetbrains.compose.desktop.application.internal
+
+import org.gradle.api.provider.Property
+import org.gradle.process.ExecOperations
+import org.jetbrains.compose.desktop.application.dsl.MacOSSigningSettings
+import java.io.*
+import java.util.regex.Pattern
+import java.util.zip.ZipEntry
+import java.util.zip.ZipInputStream
+import java.util.zip.ZipOutputStream
+
+internal class MacJarSignFileCopyingProcessor(
+    private val tempDir: File,
+    private val execOperations: ExecOperations,
+    macBundleID: Property<String?>,
+    signSettings: MacOSSigningSettings,
+) : FileCopyingProcessor {
+    private val bundleId = macBundleID.orNull
+    private val keychainPath = signSettings.keychain
+    private val signPrefix: String
+    private val signKey: String
+
+    init {
+        check(currentOS == OS.MacOS) { "$currentOS is not compatible with ${this::class.java}" }
+        check(bundleId != null) {
+            "Signing requires to specify unique application's identifier  (e.g. com.mycompany.myapp). Specify an identifier using DSL like so:\n" +
+            """|nativeExecutables {
+               |  macOS {
+               |      bundleID = "com.mycompany.myapp"
+               |  }
+               |bundleID may only contain alphanumeric characters (A-Z,a-z,0-9), hyphen (-) and period (.) characters}
+               |""".trimMargin()
+        }
+        check(bundleId.matches("[A-Za-z0-9\\-\\.]+".toRegex())) {
```

Comment:
Likely number, hyphen and dot cannot be the first character?

Reply:
I don't see such restrictions in Apple's documentation: https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleidentifier